src/metadataGeneration/tsoa.ts

```diff
@@ -157,7 +157,7 @@ export namespace Tsoa {
    */
   export interface EnumType extends TypeBase {
     dataType: 'enum';
-    enums: Array<string | number>;
+    enums: Array<string | number | boolean>;
```

Comment:
Please keep the type the same as it was because enums can not have `boolean` as a value type. See [this playground](http://www.typescriptlang.org/play/index.html#code/KYDwDg9gTgLgBMAdgVwLZwJIGcvOAZgCYAGAURAENUwAbYOAbwFgAoOduCgZRigEtEAczgBeOACJuvAYPEAaVh04A5NACNgUUXEIK2HNRAh0KiLAEEowAEIUAJtt55WAXyA) which shows that you would get `"Computed values are not permitted in an enum with string valued members."` if you tried.

Reply:
See https://github.com/lukeautry/tsoa/pull/508#issuecomment-539665332

Reply:
I think it will be confusing for maintainers.

> This is not relevant here I believe, since we are not talking about ts enums, which should be refEnums, but swagger/openapi enums, which allow booleans. Or am I missing something here?

So I’m okay with the enum in Swagger.d.ts being described as having a Boolean in there. But it doesn’t make sense for the places where a TypeScript enum is parsed into tsoa metadata.

I have a feeling that you’re using the swagger enum field as a place to store aliases. But I find that incredibly confusing.

I have to think about this some more.

Reply:
At least I'm not alone :P
The way we (ab)use the extractEnum flag to [check if we support certain constructs](https://github.com/lukeautry/tsoa/blob/master/src/metadataGeneration/parameterGenerator.ts#L193) is incredibly confusing. (See the references to the linked method)

Reply:
I can only speculate the reasoning here is to see if we can transform the parameter try into a string, which are the only valid param types in Header, Query and Path.
There, it makes sense to only allow strings and not even numbers.

Reply:
I think since we allow numbers, I don't really see why booleans wouldn't be allowed either, but I am fine with casting keeping the casts fo String for now as well, but we should refactor this for 3.0, it's confusing and buggy.

Reply:
Agreed. I don’t think we should cast a type to string.

Reply:
Also, I’d like to get rid of `extractEnum` so if you can refactor that out, that would be amazing! :)

Reply:
I already did, I am at a point where my cross dependencies are growing though.

Turns out determining the correct properties for 
```typescript
export interface TestSubModel2 extends TestSubModel, Partial<Omit<UserResponseModel, 'id'>> {
  testSubModel2: boolean;
}
```
requires quite a few steps.

Reply:
https://github.com/lukeautry/tsoa/pull/508/commits/5bded384ea75b0af32d5c59629e4f8fddef01a89

I'm not gonna even pretend I like this, but I guess this is the most compatible way to address this.

Reply:
@WoH in light of how busy I will be in the following weeks (possibly 2-3 months), I’m just gonna let you do it how you want. So please revert the last commit and just add a JSDoc comment above `enums` that explains what you said about how Swagger supports enums with Booleans.

Reply:
@dgreene1 I think it's fine to keep the implementation as compatible as possible. Maybe I voiced my concerns a bit too much. I fully understood your reasoning, I just don't think we should cherry-pick 5bded38 into 3.0 where we have an opportunity to do things better.

Reply:
Okay, but in your 3.0 branch, can you add a JSDoc comment that explains that OpenAPI supports enums with booleans? I'd ask you to link to the OpenAPI spec that explains that but I can't actually find it. As for this PR, I'll merge it.

Reply:
```yaml
components:
  schemas:
    EnumIndexValue:
      type: boolean
      enum:
        - false
```

Is valid as per JSON Schema draft 0, which Swagger & OpenAPI build on top, so they probably just refer you to this spec.
https://json-schema.org/specification-links.html#draft-0
https://tools.ietf.org/html/draft-zyp-json-schema-00#section-5.16

Reply:
I've made a separate ticket for just this: https://github.com/lukeautry/tsoa/issues/510